aesara/tensor/random/basic.py

```diff
@@ -527,8 +549,12 @@ def rng_fn(cls, rng, n, p, size):
             size = tuple(size or ())
 
             if size:
-                n = np.broadcast_to(n, size + n.shape)
-                p = np.broadcast_to(p, size + p.shape)
+                if 0 < p.ndim - 1 <= len(size) and size[-p.ndim + 1 :] != p.shape[:-1]:
```

Comment:
Do we need that check for all multivariate RVs? Is this check independent of the change of how `size` gets interpreted?

Reply:
Yes, we need to check that, because `size` and the overlapping parts of the parameters (i.e. the "independent parameter" dimensions) may not be equal, and&mdash;in some cases&mdash;that can only be determined at runtime.